v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -403,30 +403,33 @@ func (r *azureDeploymentReconcilerInstance) handleDeletePollerFailed(err error)
 	return err
 }
 
-func (r *azureDeploymentReconcilerInstance) skipReconcile(ctx context.Context) error {
-	err := r.updateStatus(ctx)
-	if err != nil {
-		if genericarmclient.IsNotFoundError(err) {
-			err = conditions.NewReadyConditionImpactingError(err, conditions.ConditionSeverityWarning, conditions.ReasonAzureResourceNotFound)
-		}
-		return err
-	}
-	return nil
-}
+type CreateOrUpdateSuccessMode string
 
-func (r *azureDeploymentReconcilerInstance) handleCreatePollerSuccess(ctx context.Context) (ctrl.Result, error) {
+const (
+	GetOnly = CreateOrUpdateSuccessMode("getOnly")
+	Full    = CreateOrUpdateSuccessMode("full")
```

Comment:
Maybe consider using a better name for these above two constants. `Full` seems like too generic, maybe fullReconcile?

Reply:
FullReconcileMode and GetOnlyReconcileMode?

Reply:
I changed to `WatchResource` or `ManageResource` to be more in line with the semantics we talk about elsewhere